src/main/java/com/arpnetworking/metrics/incubator/PeriodicMetricsFactory.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2017 Inscope Metrics, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.incubator;
+
+import com.arpnetworking.metrics.Metrics;
+
+import java.util.function.Consumer;
+
+/**
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public interface PeriodicMetricsFactory {
+    /**
+     * Register a consumer to be polled periodically.
+     *
+     * @param consumer A consumer to call to get metrics
+     */
+    void registerPolledMetric(Consumer<Metrics> consumer);
+
+    /**
+     * Get a reference to a {@link Metrics} object that will be periodically
+     * closed and refreshed.
+     *
+     * @param consumer a {@link Consumer}
+     */
+    void record(Consumer<Metrics> consumer);
+}
```

Comment:
As discussed looking for a `createMetrics` to cover the third use case of object-oriented periodic metrics access.

e.g. on initialization
`
final Metrics metrics = pmf.createMetrics();
this.myCounter = metrics.createCounter("count_something");
`

e.g. then sometime later

`
this.myConuter.increment();
`